templates/cosmic-settings.tera

```diff
@@ -0,0 +1,132 @@
+---
+name: "Catppuccin"
+window_hint_color: ""
+bg_alpha: 1.0
+frosted: false
+outer_gap_size: 0
+inner_gap_size: 8
+active_hint_size: 3
+whiskers:
+    version: 2.5.1
+    matrix:
+    - variant: ["", "round", "slightlyround", "square"]
+    - flavor
+    - accent
+    filename: "themes/{{ name }}{{ variant | capitalize }}-{{ flavor.identifier | capitalize }}-{{ accent | capitalize }}.ron"
+---
+{%- if bg_alpha < 0 or bg_alpha > 1 -%}{{ throw(message="bg_alpha must be between 0.0 and 1.0") }}{%- endif -%}
+{%- set window_hint = if(cond=window_hint_color != "", t=window_hint_color, f=accent) -%}
+{%- macro palette_color(name, color) -%}
+        {{ name }}: (
+            red: {{ color.rgb.r / 255 | trunc(places=8) }},
+            green: {{ color.rgb.g / 255 | trunc(places=8) }},
+            blue: {{ color.rgb.b / 255 | trunc(places=8) }},
+            alpha: 1.0,
+        ),
+{%- endmacro palette_color -%}
+{%- macro other_color(name, color, alpha=1.0) -%}
+    {{ name }}: Some((
+        red: {{ color.rgb.r / 255 | trunc(places=8) }},
+        green: {{ color.rgb.g / 255 | trunc(places=8) }},
+        blue: {{ color.rgb.b / 255 | trunc(places=8) }},
+        alpha: {{ alpha | trunc(places=8) }},
+    )),
+{%- endmacro other_color -%}
```

Comment:
I would also rename these macros, the current names here are very vague.

Reply:
What would a better name for them be?

Reply:
You tell me! I don't know what the underlying difference is between these two formats. Seeing as it is a Ron file, and one of them is wrapped in Some(), maybe have the second one be `optional_color`? There is also the difference of the second one including an alpha value. Can both have an alpha value or is it just the optional color?

Reply:
Well, I wrote one of them to fill in the colors in the palette section, which are always alpha 1.0, and are not optional, and the other fills in various other colors, all of which are optional (but this macro fills them in as `Some`) and one of which has an alpha value which can be changed. I couldn't think of a more descriptive name for these, so I used the name from the `generate.py`, which simply lists these colors in [`other_map`](https://github.com/catppuccin/cosmic-desktop/blob/11231f34a5de36e09a4f0fb7771a932969f3806e/generate.py#L148). `optional_color` isn't a bad name though.